heartbeat/monitors/monitor.go

```diff
@@ -151,6 +139,10 @@ func newMonitorUnsafe(
 		m.stdFields.ID = fmt.Sprintf("auto-%s-%#X", m.stdFields.Type, hash)
 	}
 
+	// De-duplicate monitors with identical IDs
+	// last write wins
+	globalDedup.register(m)
```

Comment:
`newMonitor` is used by `CheckConfig`. Checking configs shouldn't stop monitors existing monitors.

Reply:
Fixed this, it is now the case that `newMonitor` is side effect free